src/hypofuzz/database.py

```diff
@@ -174,6 +177,20 @@ def from_json(encoded: bytes, /) -> Optional["Report"]:
             return None
 
 
+@dataclass(frozen=True)
+class ReportWithDiff(Report):
+    # This fields are not stored in the database, but are computed for an
+    # individual Report, relative to another Report.
+    status_counts_diff: StatusCounts
+    elapsed_time_diff: float
+    # The timestamp of consecutive reports is not always monotonic - due to
+    # daylight savings time, for example, or merely NTP clock updates. We therefore
+    # store a separate timestamp_monotonic, which uses timestamp if the timestamp
+    # is monotonic relative to the previous report, and otherwise
+    # previous_report.timestamp_monotonic + elapsed_time_diff.
+    timestamp_monotonic: float
```

Comment:
- I misspoke; daylight savings doesn't affect unix time, though NTP updates (or manual clock adjustments) do.
- it's not monotonic _relative to previous report_, but rather prev_report + elapsed_time diff.

```suggestion
    # Clock updates due to e.g. NTP can make time.time() non-monotonic, so
    # to preserve ordering in edge cases we define `timestamp_monotonic` as
    # `min(time.time(), previous_timestamp_monotonic + elapsed_time_diff)`.
    timestamp_monotonic: float
```

Reply:
(max, right, not min?)